app/lib/dartdoc/backend.dart

```diff
@@ -187,7 +187,18 @@ class DartdocBackend {
       if (hasServing) {
         completedList.removeWhere((entry) => !entry.isServing);
       }
-      completedList.sort((a, b) => -a.timestamp.compareTo(b.timestamp));
+      completedList.sort((a, b) {
+        // content goes first
+        if (a.hasContent && !b.hasContent) return -1;
+        if (!a.hasContent && b.hasContent) return 1;
+        // next release goes to the back of the list
+        final rva = a.runtimeVersion.compareTo(shared_versions.runtimeVersion);
+        final rvb = b.runtimeVersion.compareTo(shared_versions.runtimeVersion);
+        if (rva > 0 && rvb <= 0) return 1;
+        if (rva <= 0 && rvb > 1) return -1;
+        // otherwise compare by timestamp descending
+        return -a.timestamp.compareTo(b.timestamp);
+      });
```

Comment:
Can I block this PR on: https://github.com/dart-lang/pub-dartlang-dart/issues/1828
I'm not quite sure what this attempts to do...

Also couldn't we do it simpler? Something along the lines of:
```dart
import '../shared/versions.dart' show runtimeVersion, semanticRuntimeVersion;

...

    Future<DartdocEntry> loadVersion(String v) async {
      final List<DartdocEntry> entries =
          await _listEntries(storage_path.entryPrefix(package, v));
      // utility to remove anything not matching pred, unless that is everything
      void prefer(pred) => entries.any(pred) ? entries.retainWhere(pred) : null;
      prefer((e) => e.isServing && e.hasContent);
      prefer((e) => e.hasContent);
      prefer((e) => e.runtimeVersion == runtimeVersion);
      entries.sort((a, b) => -a.timestamp.compareTo(b.timestamp));
      return entries.isEmpty ? null : entries.first;
    }
```

Reply:
If possible can we at-least avoid hacking with sort... sorting by timestamp is fine... but mixing in `hasContent` and `runtimeVersion` makes this very complicated..